### ISSUE TITLE
Repeat last answer for bicycle lane quests

### ISSUE THREAD

**Issue by debuglevel** [streetcomplete/StreetComplete#2542]
**Use case**
On some roads, there are really many bicycle lane quests. I walked a street about 200meters long and answered 8 times: "no, there is no bicycle lane", "no, again not", "still not", "nope, nothing changed", "no, still the same road without any bicycle lane", ... (and that for both sides). That is about 6 taps for 1 quest.

**Proposed Solution**
- Most simple solution would be "repeat my last answer".
- A more sophisticated solution might be to predict (if there is no data available) the bicycle lane situation for the next quests. If I answered "no bicycle lanes here on both sides" on the beginning of a street, it might be sane to suggest "no bicycle lanes on both sides" for the rest of the road (at least I could just tap OK then). If anything changes, I can still change my answer for the next bit of the street.

**Comment by andrewharvey** [comment]
another idea is to be able to select the middle of the road in the graphic to apply the same value to both sides (very common for none).

**Comment by debuglevel** [comment]
@andrewharvey yeah, that would have been my next issue :-). But then I thought "if i can repeat my last answer" (which often is "none on both sides" in Germany), that would also be okay.

**Comment by westnordost** [comment]
Well, as for the interface I'd find something similar to the building levels quest consistent: A little button to the lower right that shows the last answer. Tapping on it prefills the form with the last answer.

**Comment by smichel17** [comment]
One complication is that I'm not sure what to do if your last answer was a cycleway on one side only. I don't think there is a way to automatically decide which orientation is correct. I could imagine having two buttons in this case, but the icons would need to rotate with the map.

On February 2, 2021 9:55:39 PM UTC, Tobias Zwick <notifications@github.com> wrote:
>Well, as for the interface I'd find something similar to the building levels quest consistent: A little button to the lower right that shows the last answer. Tapping on it prefills the form with the last answer.
>

-- 
To respect your time, I try to write short, functional emails.

**Comment by arrival-spring** [comment]
> One complication is that I'm not sure what to do if your last answer was a cycleway on one side only. I don't think there is a way to automatically decide which orientation is correct. I could imagine having two buttons in this case, but the icons would need to rotate with the map.

How about there's the little bubble like the building levels quest and then when you tap it it becomes a double sided arrow which swaps the sides of the lanes?

**Comment by Lee-Carre** [comment]
This would be most useful for me. Roads, here, are often too small to have cycleways. Instead, there are several explicit cycle routes.

Much like for adding building:levels and roof:levels for a series of identical houses, a UI button (or other widget, but consistency between quest types would be good) to input the MRU values would aid efficiency.

Better yet, perhaps a small drop-down with the most _frequently_ used values (imagine needing to alternate between two different answers for the vast majority of roads).

**Comment by smichel17** [comment]
Related: https://github.com/streetcomplete/StreetComplete/discussions/2684 for making this quest faster to answer
